ironfish/src/mining/manager.test.slow.ts

```diff
@@ -546,6 +546,94 @@ describe('Mining manager', () => {
       expect(node.memPool.count()).toBe(1)
     })
 
+    it('should not create block templates with invalid evm transactions', async () => {
+      const { node, chain } = nodeTest
+      const { miningManager } = node
+
+      jest
+        .spyOn(node.chain.consensus, 'getActiveTransactionVersion')
+        .mockImplementation(() => TransactionVersion.V3)
+
+      const account = await useAccountFixture(nodeTest.node.wallet, 'account')
+      await nodeTest.node.wallet.setDefaultAccount(account.name)
+
+      const previous = await useMinerBlockFixture(chain, 2, account, node.wallet)
+      await expect(chain).toAddBlock(previous)
+      await node.wallet.scan()
+
+      const ethAddress = Address.fromPrivateKey(Buffer.from(account.spendingKey, 'hex'))
+      const ethAccount = new EthAccount(0n, 1000n)
+
+      const evmRecipientAddress = Address.fromString(
+        '0x3c6c6d51f71a0f146cf79843e888feb20258f567',
+      )
+
+      await node.chain.blockchainDb.transaction(async (_) => {
+        await node.chain.blockchainDb.stateManager.putAccount(ethAddress, ethAccount)
+      })
+
```

Comment:
Interesting question, what if a new account submits two transactions at once, but the mining manager tries to add the one with a higher nonce first?

Reply:
In that case I think the one with the higher nonce would be skipped as invalid (assuming the higher nonce doesn't match the account nonce)

Reply:
Ahh yes you are right!